pkg/provider/onboardbase/client/client.go

```diff
@@ -0,0 +1,349 @@
+/*
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package client
+
+import (
+	"bytes"
+	"crypto/tls"
+	"encoding/json"
+	"fmt"
+	"io"
+	"net/http"
+	"net/url"
+	"strings"
+	"time"
+
+	aesdecrypt "github.com/Onboardbase/go-cryptojs-aes-decrypt/decrypt"
+)
+
+type OnboardbaseClient struct {
+	baseURL      *url.URL
+	OnboardbaseAPIKey string
+	VerifyTLS    bool
+	UserAgent    string
+	OnboardbasePassCode string
+	httpClient *http.Client
+}
+
+type queryParams map[string]string
+
+type headers map[string]string
+
+type httpRequestBody []byte
+
+type Secrets map[string]string
+
+type RawSecret struct {
+	Key string `json:"key,omitempty"`
+	Value string `json:"value,omitempty"`
+}
+
+type RawSecrets []RawSecret
+
+type APIError struct {
+	Err     error
+	Message string
+	Data    string
+}
+
+type apiResponse struct {
+	HTTPResponse *http.Response
+	Body         []byte
+}
+
+type apiErrorResponse struct {
+	Messages []string
+	Success  bool
+}
+
+type SecretRequest struct {
+	Environment    string
+	Project string
+	Name    string
+}
+
+type SecretsRequest struct {
+	Environment    string
+	Project string
+}
+
+type UpdateSecretsRequest struct {
+	Secrets RawSecrets `json:"secrets,omitempty"`
+	Project string     `json:"project,omitempty"`
+	Config  string     `json:"config,omitempty"`
+}
+
+type secretResponseBodyObject struct {
+	Title string `json:"title,omitempty"`
+	Id string `json:"id,omitempty"`
+}
+
+type secretResponseBodyData struct {
+	Project secretResponseBodyObject `json:"project,omitempty"`
+	Environment secretResponseBodyObject `json:"environment,omitempty"`
+	Team secretResponseBodyObject `json:"team,omitempty"`
+	Secrets []string `json:"secrets,omitempty"`
```

Comment:
This structure here does not reflect the payload response provided by onboard-base. It simply means it does not work.